vignettes/Data_retrieval_via_API.Rmd

```diff
@@ -0,0 +1,507 @@
+printFormattedTable(unit)
+```
+
+## Load sample/chemical information
+
+### Load concentration unit
+
+Use <font face="CMTT10">tcplLoadConcUnit</font> to load the concentration unit for a specific `spid` or multiple `spid`s.
+
+```{r load_conc_unit}
+# load conc_unit for given spid
+conc_unit <- tcplLoadConcUnit(spid = "TP0000904H05")
+```
+
+```{r, echo = FALSE}
+printFormattedTable(conc_unit)
+```
+
+### Load chemical info
+
+Use <font face="CMTT10">tcplLoadChem</font> to load the chemical information for a specific `spid` or multiple `spid`s. Notice tcplLoadChem uses `field` instead of `fld`.
+
+```{r load_chem}
+# load chem_info for given spid
+chem_info <- tcplLoadChem(field = "spid", val = "TP0000904H05")
+```
+
+```{r, echo = FALSE}
+printFormattedTable(chem_info)
+```
+
+# Plotting
+
+<font face="CMTT10">tcplPlot</font> is tcpl’s single flexible plotting function, allowing for interactive and consistent visualization of concentration-response curves via customizable parameters. For more details on implementation, parameters and specific customization instructions, refer to the main Data Retrieval vignette. This section will instead focus on the limitations of using the CCTE API as a data source for <font face="CMTT10">tcplPlot</font>.
+
+Similar to tcplLoadData, the user must indicate the applicable fields and ids which are query-able via the CCTE API. These such fields include and are limited to "AEID", "m4id", "SPID", and "DTXSID". Any other fields will result in error. While supplying multiple ids (through `val`) are valid, multiple fields are not, so combinations of fields are not compatible with <font face="CMTT10">tcplPlot</font> when connected to the API. For example, if `fld = "spid"`, no aeid can be specified, meaning every matching spid will be plotted. If `fld = "aeid"`, every sample within the given endpoint(s) will be plotted. If `fld = "m4id"`, only one plot will output for every each m4id input. Therefore, if looking for a specific aeid/spid combo, one should determine the corresponding m4id, like below.
+
+```{r plot_aeid_spid, eval = FALSE}
+# load all matching spids and then subset using the aeid desired to find m4id
+mc5 <- tcplLoadData(lvl = 5, 
+                    fld = "spid",
+                    val = "TP0000904H05",
+                    type = "mc", 
+                    add.fld = FALSE) # 8 rows of data
+m4id <- mc5[aeid == 714]$m4id # subset to 1 aeid extract m4id
+
+# default parameters used here: fld = "m4id", type = "mc" (type can never be "sc" when connected to API)
+tcplPlot(val = m4id, output = "jpg", verbose = TRUE, flags = TRUE)
+```
+
+<center>![<font style="font-size:15px"><i>API-sourced plot with parameters: output = "jpg", verbose = TRUE, and flags = TRUE for m4id 1847540</i></font>](img/API_plot_1847540.jpg)</center>
+
+Use the "compare" feature to supply more m4ids to `compare.val` and compare them one to one with the m4ids in `val`. `val` and `compare.val` must be the same length!
```

Comment:
The “compare” parameter can be used by supplying a list of m4ids as val and compare.val. These lists must be the same length!